src/Infrastructure/Repositories/LocalFilesRepository.php

```diff
@@ -42,84 +42,107 @@ public function getDefaultDirectory(): string
         return (string) getcwd();
     }
 
+    /**
+     * @return array<string, \Symfony\Component\Finder\SplFileInfo>
+     */
     public function getFiles(): array
     {
-        if ($this->files === null) {
-            $this->files = $this->getFilesList();
+        if (count($this->files) !== 0) {
+            return $this->files;
         }
 
-        return $this->files;
+        $withPathInfo = static function (string $path): array {
+            return [
+                'dirname' => $dirname = pathinfo($path, PATHINFO_DIRNAME),
+                'basename' => $basename = pathinfo($path, PATHINFO_BASENAME),
+                'full_path' => $dirname . DIRECTORY_SEPARATOR . $basename,
+                'is_file' => ! is_dir($path) && is_file($path),
+            ];
+        };
+
+        $paths = array_map($withPathInfo, $this->paths);
+
+        $directories = array_filter($paths, fn (array $path): bool => ! $path['is_file']);
+        $files = array_filter($paths, fn (array $path): bool => $path['is_file']);
+
+        return $this->files = array_merge(
+            $this->filesWithinDirectories($directories, $this->exclude),
```

Comment:
As we store now $exclude in `$this->exclude`, we may drop the second arg, right ?

Reply:
My personal preference is to keep methods as stateless as possible, because by accessing state from one method and passing it through to other methods explicitly we reduce the dependency each method has on the structure of the parent. That said, I'm happy either way: if you think it's more maintainable / readable / understandable to access `$this->exclude` directly then I am happy to make that change! :) I'll do that now.

Reply:
You convinced me ! Let's leave this like this